packages/app/src/state/savedGraphs.ts

```diff
@@ -144,56 +126,35 @@ export type OpenedProjectsInfo = {
   openedProjectsSortedIds: ProjectId[];
 };
 
-export const projectsState = atom<OpenedProjectsInfo>({
-  key: 'projectsState',
-  default: {
+export const projectsState = atomWithStorage<OpenedProjectsInfo>(
+  'projectsState',
+  {
     openedProjects: {},
     openedProjectsSortedIds: [],
   },
-  effects: [persistAtom],
-});
-
-export const openedProjectsState = selector({
-  key: 'openedProjectsState',
-  get: ({ get }) => {
-    return get(projectsState).openedProjects;
-  },
-  set: ({ set }, newValue) => {
-    set(projectsState, (oldValue) => {
-      return {
-        ...oldValue,
-        openedProjects: newValue instanceof DefaultValue ? {} : newValue,
-      };
+  storage,
+);
+
+export const openedProjectsState = atom(
+  (get) => get(projectsState).openedProjects,
+  (get, set, newValue: Record<ProjectId, OpenedProjectInfo> | undefined) => {
+    set(projectsState, {
+      ...get(projectsState),
+      openedProjects: newValue ?? {},
     });
   },
-});
-
-export const openedProjectsSortedIdsState = selector({
-  key: 'openedProjectsSortedIdsState',
-  get: ({ get }) => {
-    return get(projectsState).openedProjectsSortedIds;
-  },
-  set: ({ set }, newValue) => {
-    set(projectsState, (oldValue) => {
-      return {
-        ...oldValue,
-        openedProjectsSortedIds: newValue instanceof DefaultValue ? [] : newValue,
-      };
+);
+
+export const openedProjectsSortedIdsState = atom(
+  (get) => get(projectsState).openedProjectsSortedIds,
+  (get, set, newValue: ProjectId[] | undefined) => {
+    set(projectsState, {
+      ...get(projectsState),
+      openedProjectsSortedIds: newValue ?? [],
     });
   },
-});
+);
 
-/** Project context values stored in the IDE and not in the project file. Available in Context nodes. */
-export type ProjectContext = Record<
-  string,
-  {
-    value: DataValue;
-    secret: boolean;
-  }
->;
-
-export const projectContextState = atomFamily<ProjectContext, ProjectId>({
-  key: 'projectContext',
-  default: {},
-  effects: [persistAtom],
-});
+export const projectContextState = atomFamily((projectId: ProjectId) =>
+  atomWithStorage<ProjectContext>(`projectContext-${projectId}`, {}, storage),
```

Comment:
@abrenneke How do you add context to a project through the UI? I need to test this out.

Not sure how the previous recoil storage keys records stored as part of an atomFamily. I assume its by projectId but I'd like to test it.